# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/crashreporter/nsExceptionHandler.cpp
# Commit: bff57b2e4b16
# Full Hash: bff57b2e4b16fbcc4b5bf881c66a352ed7e2bd92
# Author: Gabriele Svelto <gsvelto@mozilla.com>
# Date: 2022-09-12 21:40:55
# Regressor Bug: 1789421
# File Overlap Count: 1
# Description:
#   Bug 1789421 - Generate crash reports for content process shutdown hangs only when we can confirm that the process did not shut down in time r=jstutte
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156838
# ==============================================================================

diff -r f9922dd645d5 -r bff57b2e4b16 toolkit/crashreporter/nsExceptionHandler.cpp
--- a/toolkit/crashreporter/nsExceptionHandler.cpp	Mon Sep 12 08:56:33 2022 +0000
+++ b/toolkit/crashreporter/nsExceptionHandler.cpp	Mon Sep 12 08:57:16 2022 +0000
@@ -3042,24 +3042,38 @@
   }
 }
 
-void DeleteMinidumpFilesForID(const nsAString& id) {
+void DeleteMinidumpFilesForID(const nsAString& aId,
+                              const Maybe<nsString>& aAdditionalMinidump) {
   nsCOMPtr<nsIFile> minidumpFile;
-  if (GetMinidumpForID(id, getter_AddRefs(minidumpFile))) {
+  if (GetMinidumpForID(aId, getter_AddRefs(minidumpFile))) {
     minidumpFile->Remove(false);
   }
 
   nsCOMPtr<nsIFile> extraFile;
-  if (GetExtraFileForID(id, getter_AddRefs(extraFile))) {
+  if (GetExtraFileForID(aId, getter_AddRefs(extraFile))) {
     extraFile->Remove(false);
   }
+
+  if (aAdditionalMinidump && GetMinidumpForID(aId, getter_AddRefs(minidumpFile),
+                                              aAdditionalMinidump)) {
+    minidumpFile->Remove(false);
+  }
 }
 
-bool GetMinidumpForID(const nsAString& id, nsIFile** minidump) {
+bool GetMinidumpForID(const nsAString& id, nsIFile** minidump,
+                      const Maybe<nsString>& aAdditionalMinidump) {
   if (!GetMinidumpLimboDir(minidump)) {
     return false;
   }
 
-  (*minidump)->Append(id + u".dmp"_ns);
+  (*minidump)->Append(id);
+
+  if (aAdditionalMinidump) {
+    (*minidump)->Append(u"-"_ns);
+    (*minidump)->Append(*aAdditionalMinidump);
+  }
+
+  (*minidump)->Append(u".dmp"_ns);
 
   bool exists;
   if (NS_FAILED((*minidump)->Exists(&exists)) || !exists) {